chore: update owner field docs (#4698)
aws-amplify/docs: src/pages/cli/graphql/authorization-rules.mdx

You can override the `owner` field to your own preferred field, by specifying a custom `ownerField` in the authorization rule.

> Do not set `ownerField` to your `@primaryKey` field or `id` field if no primary key is specified. If you want to query by the `ownerField`, use an `@index` on that `ownerField` to create a secondary index.

Reviewer: ```suggestion
<Callout>

Do not set `ownerField` to your `@primaryKey` field or `id` field if no primary key is specified. If you want to query by the `ownerField`, use an `@index` on that `ownerField` to create a secondary index.

</Callout>
```

> It is not recommended to set your `ownerField` to be the same as your `@primaryKey` field, or `id` if no primary key is specified. If you wish to be able to query by the `ownerField` specified, using an `@index` on that `ownerField` is recommended instead.

Reviewer 2: It'd be great if this had a bit more context on why—it's not quite clear to me why this isn't a valid pattern, per https://github.com/aws-amplify/amplify-category-api/issues/872#issuecomment-1278488889

Reviewer 3: @alharris-at - Wanted to bump this to see if the conversation can be resolved. Thank you!

> It is not recommended to set your `ownerField` to be the same as your `@primaryKey` field, or `id` if no primary key is specified. If you wish to be able to query by the `ownerField` specified, using an `@index` on that `ownerField` is recommended instead.

Reviewer 4: Not a blocker:

```suggestion
<Callout>

It is not recommended to set your `ownerField` to be the same as your `@primaryKey` field, or `id` if no primary key is specified. If you wish to be able to query by the `ownerField` specified, using an `@index` on that `ownerField` is recommended instead.

</Callout>
```

> It is not recommended to set your `ownerField` to be the same as your `@primaryKey` field, or `id` if no primary key is specified. If you wish to be able to query by the `ownerField` specified, using an `@index` on that `ownerField` is recommended instead.

Reviewer: ```suggestion
Do not set `ownerField` to your `@primaryKey` field or `id` field if no primary key is specified. If you want to query by the `ownerField`, use an `@index` on that `ownerField` to create a secondary index.
```
